libs/@hashintel/type-editor/src/entity-type-editor/property-list-card.tsx

```diff
@@ -393,7 +407,12 @@ export const PropertyTypeRow = ({
 }) => {
   const { control } = useFormContext<EntityTypeEditorFormData>();
 
-  const array = useWatch({
+  const isRequired = useWatch({
+    control,
+    name: `properties.${propertyIndex}.required`,
+  });
+
+  const isArray = useWatch({
     control,
     name: `properties.${propertyIndex}.array`,
   });
```

Comment:
You can do this with one `useWatch`, like so

```ts
const [isRequired, isArray] = useWatch({ control, name: [`properties.${propertyIndex}.required`, `properties.${propertyIndex}.array`] })
```

Reply:
Thank you – done in https://github.com/hashintel/hash/pull/2117/commits/32cf493ef485387fe504aae335a9971ad8bdb49a